### ISSUE TITLE
patch-package for attributes and uid (example from issues)

### ISSUE THREAD

**Issue by Liiion911** [MateMalice/mail-listener2#28]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `mail-listener5@2.0.2` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/mail-listener5/index.js b/node_modules/mail-listener5/index.js
index a984b83..7ceeb03 100644
--- a/node_modules/mail-listener5/index.js
+++ b/node_modules/mail-listener5/index.js
@@ -107,11 +107,15 @@ class MailListener extends EventEmitter {
             markSeen: self.markSeen
           });
           f.on('message', (msg, seqno) => {
+            let attrs;
+            msg.on('attributes', (a) => {
+              attrs = a;
+            });
             msg.on('body', async (stream, info) => {
               let parsed = await simpleParser(stream);
-              self.emit('mail', parsed, seqno);
-              self.emit('headers', parsed.headers, seqno);
-              self.emit('body', {html: parsed.html, text: parsed.text, textAsHtml: parsed.textAsHtml}, seqno);
+              self.emit('mail', parsed, seqno, attributes);
+              self.emit('headers', parsed.headers, seqno, attributes);
+              self.emit('body', {html: parsed.html, text: parsed.text, textAsHtml: parsed.textAsHtml}, seqno, attrs);
               if (parsed.attachments.length>0)
               {
                 for (let att of parsed.attachments)
@@ -121,11 +125,11 @@ class MailListener extends EventEmitter {
                     await fs.writeFile(`${self.attachmentOptions.directory}${att.filename}`, att.content, (error) =>{
                       self.emit('error', error);
                     });
-                    self.emit('attachment', att, `${self.attachmentOptions.directory}${att.filename}`, seqno);
+                    self.emit('attachment', att, `${self.attachmentOptions.directory}${att.filename}`, seqno, attrs);
                   }
                   else
                   {
-                    self.emit('attachment', att, null, seqno);
+                    self.emit('attachment', att, null, seqno, attrs);
                   }
                 }
               }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by MateMalice** [comment]
Thank you for the comment - from looking at the code, this may fix an issue I've noticed with the handling of attributes.

If you open a pull request with your suggested changes, then I'll test it & may merge it with the main branch. :-)

**Comment by MateMalice** [comment]
Fixed in [v2.1.1](https://github.com/MateMalice/mail-listener2/releases/tag/v2.1.1).
